source-academy/frontend
# Ensure that editor tabs have corresponding files

### Description

Resolves the following issue that was raised by @zhaojj2209 in #2378:

> - When first turning on multiple files mode, `program.js` is not visible in the sidebar. The file will only appear in the sidebar after typing in some code and then reloading the page.
> ![telegram-cloud-photo-size-5-6104838078012175334-y](https://user-images.githubusercontent.com/54243224/226177102-678a08c1-7dee-46f3-be3c-d25ce665a814.jpg)

Also centralise the configuration of base paths of each workspace for better cohesion.

Part of #2176.

### Type of change


### How to test

The multiple files mode is not yet ready for production and is disabled programmatically. Enable multiple files mode for testing by changing the condition to `false`:
https://github.com/source-academy/frontend/blob/9bc11408b104e29679bfdc025992b241975ea6d7/src/pages/playground/Playground.tsx#L603-L615

It might be easier to verify this by opening an incognito tab in Chrome. Note that this does not work in Firefox where IndexedDB access is entirely blocked in private browsing mode.
